app/controllers/settings_controller.rb

```diff
@@ -120,8 +121,43 @@ def update_impact_stats
     render json: { message: 'Impact Stats Updated Successfully.' }, status: :ok
   end
 
+  def fetch_site_notice
+    render json: { site_notice_env: ENV['sitenotice'],
+                   current_site_notice: current_site_notice['message'] }, status: :ok
+  end
+
+  def update_site_notice
+    updated_site_notice = params[:siteNotice]
+    Setting.set_hash('site_notice', 'message', updated_site_notice)
+    Rails.cache.delete('site_notice')
+    set_site_notice unless ENV['sitenotice'].blank?
+    render json: { message: 'Site Notice Updated Successfully.' }, status: :ok
+  end
+
+  def toggle_site_notice
+    if ENV['sitenotice'].blank?
+      set_site_notice
+      message = 'Site Notice Enabled Successfully.'
+    else
+      ENV['sitenotice'] = ''
+      message = 'Site Notice Disabled Successfully.'
+    end
+    render json: { message: }, status: :ok
+  end
+
   private
 
+  def set_site_notice
+    ENV['sitenotice'] = current_site_notice['message']
```

Comment:
Will this strategy retain the site notice across application or server restarts? I was expecting this to require some minor changes for how the message gets served to the frontend, such as going through the `Features` presenter in `_flash.htmlhaml` instead of using the ENV.

Reply:
you are right it won't retain site notice when server restarts. Now i have changed the approach using Features presenter instead of ENV to provide sitenotice from cache/database to _flash.htmlhaml